lib/Server.js

```diff
@@ -382,12 +382,23 @@ class Server {
               }
 
               const bypass = typeof proxyConfig.bypass === 'function';
-
+              const bypassRawValue = bypass
+                ? proxyConfig.bypass(req, res, proxyConfig)
+                : null;
               const bypassUrl =
-                (bypass && proxyConfig.bypass(req, res, proxyConfig)) || false;
+                typeof bypassRawValue === 'string'
+                  ? bypassRawValue
+                  : typeof bypassRawValue === 'boolean' &&
+                    bypassRawValue === false;
```

Comment:
The type check is superfluous because the triple (exactly) equal already asserts the type check.

Reply:
this plus the `if` below is massively convoluted. please don't do this. also code comments 😳